core/domain/topic_services.py

```diff
@@ -255,15 +258,29 @@ def apply_change_list(topic_id, change_list):
             topic.
 
     Returns:
-        Topic. The resulting topic domain object.
+        dict, list(int), list(int). A dict with two fields: topic and
```

Comment:
Maybe split first dict into a Topic domain object and a list of SubtopicPage domain objects. The latter could represent modified_subtopic_pages -- you don't need to fetch every subtopic page.

Otherwise the first return value here is a somewhat arbitrary dict with an opaque structure. Better to split it into well-defined objects.

Reply:
So, you mean to only fetch the subtopic pages that are to be modified? So, won't that mean the subtopic page fetch should be done when doing each change (provided the page wasn't already fetched)?